### PR TITLE
Log Unmatch and Errors

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
Seems like a little improvement to request logging.

@jmcardon Is going to love this one

### CODE REVIEW

**Comment by ChristopherDavenport** [comment]
Scalafmt hates me!

**Comment by ChristopherDavenport** [comment]
I lean towards 2. Where we explicitly log that it went through the middleware. and was an error/unmatched and that the behavior is whatever the expected behavior outside is.

File: server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala [hunk 1]
```diff
@@ -42,15 +42,35 @@ object RequestLogger {
             )
 
             val response = routes(changedRequest)
-            response.map { resp =>
-              resp.withBodyStream(
-                resp.body.onFinalize(
+            response.attempt
+              .flatMap {
+                case Left(e) =>
+                  OptionT.liftF(
```
Comment (jmcardon) [inline server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala]: I think this entire thing can just be one big `OptionT.liftF(logyourbleepblorp >> F.raiseError(e))`  if we want to be pedantic.

File: server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala [hunk 2]
```diff
@@ -42,15 +42,35 @@ object RequestLogger {
...
+                        Logger.logMessage[F, Request[F]](req.withBodyStream(newBody))(
+                          logHeaders,
+                          logBody,
+                          redactHeadersWhen)(logger.info(_))
+                      )
+                    )
+                  )
+              }
+              .orElse(
+                OptionT.liftF(
```
Comment (jmcardon) [inline server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala]: This can probable be OptionT(log >> F.pure(None)).

**Pull request merged**